packages/typegpu/src/tgsl/wgslGenerator.ts

```diff
@@ -168,13 +168,16 @@ export function generateExpression(
     const rhsExpr = generateExpression(ctx, rhs);
 
     const forcedType = expression[0] === NODE.assignmentExpr
-      ? [lhsExpr.dataType as AnyData]
-      : [];
+      ? lhsExpr.dataType.type === 'ptr'
+        ? [lhsExpr.dataType.inner as AnyData]
+        : [lhsExpr.dataType as AnyData]
+      : undefined;
 
     const converted = convertToCommonType(
       ctx,
       [lhsExpr, rhsExpr],
       op === '/' ? [f32, f16] : forcedType,
+      op !== '/',
```

Comment:
Could we add a comment here saying which parameter it is?
```suggestion
      /* verbose */ op !== '/',
```